apis/v1alpha2/gateway_types.go

```diff
@@ -663,19 +651,24 @@ const (
 
 // ListenerStatus is the status associated with a Listener.
 type ListenerStatus struct {
-	// Port is the unique Listener port value for which this message is
-	// reporting the status.
-	Port PortNumber `json:"port"`
-
-	// Protocol is the Listener protocol value for which this message is
-	// reporting the status.
-	Protocol ProtocolType `json:"protocol"`
+	// Name is the name of the Listener.
+	//
+	// +kubebuilder:validation:MaxLength=253
+	Name string `json:"name,omitempty"`
```

Comment:
I think that this field needs to be required, or else it is not possible to disambiguate listener status.